server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -654,6 +655,26 @@ public final Boolean readOptionalBoolean() throws IOException {
         return map;
     }
 
+    /**
+     * Reads a {@link List} of {@code V}-type and then converts them to a {@code Map} using keyMapper.
+     *
+     * @param valueReader The value reader
+     * @param keyMapper function to create a key from a value
+     * @return Never {@code null}.
+     */
+    public <K, V> Map<K, V> readMapFromList(final Writeable.Reader<V> valueReader, final Function<V, K> keyMapper) throws IOException {
+        final int size = readArraySize();
+        if (size == 0) {
+            return Collections.emptyMap();
+        }
+        final Map<K, V> map = Maps.newMapWithExpectedSize(size);
+        for (int i = 0; i < size; ++i) {
```

Comment:
I believe the idiomatic way is to use post-increment `i++` albeit it doesn't make any difference